### PR TITLE
8313081: MonitoringSupport_lock should be unconditionally initialized after 8304074

### PR DESCRIPTION
**Pull request by phohensee**
Follow-on bug fix for [JDK-8304074](https://bugs.openjdk.org/browse/JDK-8304074) backport. Clean backport from 17u, see https://github.com/openjdk/jdk17u-dev/pull/1981. The lock definition macro changed between 17 and 21, so the definition of MonitoringSupport_lock is different.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] [JDK-8313081](https://bugs.openjdk.org/browse/JDK-8313081) needs maintainer approval
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313081](https://bugs.openjdk.org/browse/JDK-8313081): MonitoringSupport_lock should be unconditionally initialized after 8304074 (**Bug** - P4 - Approved)

### Reviewers
 * [Volker Simonis](https://openjdk.org/census#simonis) (@simonis - **Reviewer**) ⚠️ Review applies to [7c283d7a](https://git.openjdk.org/jdk11u-dev/pull/2287/files/7c283d7ad9b38227d4095f7e34767e9ef0b91eea)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2287/head:pull/2287` \
`$ git checkout pull/2287`

Update a local copy of the PR: \
`$ git checkout pull/2287` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2287/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2287`

View PR using the GUI difftool: \
`$ git pr show -t 2287`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2287.diff">https://git.openjdk.org/jdk11u-dev/pull/2287.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2287#issuecomment-1821360654)

### CODE REVIEW

File: test/jdk/com/sun/management/ThreadMXBean/ThreadAllocatedMemory.java [hunk 1]
```diff
@@ -22,10 +22,19 @@
  */
 
 /*
- * @test
- * @bug     6173675 8231209 8304074
+ * @test    id=G1
+ * @bug     6173675 8231209 8304074 8313081
  * @summary Basic test of ThreadMXBean.getThreadAllocatedBytes
- * @author  Paul Hohensee
+ * @requires vm.gc == "G1"
```
Comment (simonis) [inline test/jdk/com/sun/management/ThreadMXBean/ThreadAllocatedMemory.java:28]: Why have you changed this from `@requires vm.gc.G1` to `@requires vm.gc == "G1"`? I think the former should also be supported in jdk11u-dev.

File: test/jdk/com/sun/management/ThreadMXBean/ThreadAllocatedMemory.java [hunk 2]
```diff
@@ -22,10 +22,19 @@
...
+ * @run main/othervm -XX:+UseG1GC ThreadAllocatedMemory
+ */
+
+/*
+ * @test    id=Serial
+ * @bug     6173675 8231209 8304074 8313081
+ * @summary Basic test of ThreadMXBean.getThreadAllocatedBytes
+ * @requires vm.gc == "Serial"
```
Comment (simonis) [inline test/jdk/com/sun/management/ThreadMXBean/ThreadAllocatedMemory.java:36]: Why have you changed this from `@requires vm.gc.Serial` to `@requires vm.gc == "Serial"`? I think the former should also be supported in jdk11u-dev.

**Review by simonis — APPROVED** [review]
Looks good but see my inline comments.

**Comment by phohensee** [comment]
/approval request Almost clean backport of a follow-on fix for the JDK-8304074 backport at https://github.com/openjdk/jdk11u-dev/pull/2286. In production at Amazon for over two months.

**Pull request closed without merging**